lineapy/plugins/jinja_templates/dvc_dag_PythonOperator.jinja

```diff
@@ -0,0 +1,18 @@
+import {{ MODULE_NAME }}
+import pickle
+
+def task_{{ STAGE["name"] }}():
+{%- if STAGE["deps"]|length > 0 %}
+    {%- for dep in STAGE["deps"] %}
+    {{ dep }} = pickle.load(open('{{ dep }}.pickle','rb'))
+    {%- endfor %}
+{%- endif %}
+{%- if STAGE["outs"]|length > 0 %}
+    {%- for out in STAGE["outs"] %}
+    {{ out }} = {{ MODULE_NAME }}.get_{{ out }}({%- if STAGE["deps"]|length > 0 %}{%- for dep in STAGE["deps"] %}{{ dep }}={{ dep }}{%- endfor %}{%- endif %})
```

Comment:
@pd-t I still think this line won't quite work. 

Can you run the housing example under https://github.com/LineaLabs/lineapy/blob/main/examples/use_cases/predict_house_price/housing_example.ipynb but replace Cell36 with DVC configurations? 

```
# NBVAL_SKIP
import os
directory = lineapy.to_pipeline(
    ["cleaned_data_housing", "linear_model_housing"],
    input_parameters=['data_url'],
    framework="DVC",
    pipeline_name="data_housing_pipeline",
    dependencies={ "linear_model_housing": { "cleaned_data_housing" } },
    pipeline_dag_config={
        "dag_flavor": "StagePerArtifact",
    },
    output_dir="./dvc/dags/"
)
```

When i try to `dvc repro` with the resulting files I get 
```
AttributeError: module 'data_housing_pipeline_module' has no attribute 'get_training_data'
``` 
which is because the module file has `def get_cleaned_data_housing(data_url):` using `cleaned_data_housing` which is some version of the artifact name which is non-conflicting, but 

the task you've generated has `data_housing_pipeline_module.get_training_data()` which uses `training_data` which the variable name. 

I think you can use the TaskDefinition.call_block safely here unless you have an example where it doesn't work?

Reply:
The call block is working fine but this only shifted the problem. The artifact function for the cleaned data in the module file expects the default input variable data_url of the session. I don't see why the BasePipelineWriter handles the variable assignment b0 in the simple unit test and the data_url assignment in the housing notebook example different. I assume, that a notebook is a session per default and a python code is not? Is there a already unit test case where this behaviour (regarding the default input variables) can be reproduced?

Reply:
![image](https://user-images.githubusercontent.com/67829287/210584428-a266061d-d905-4098-a906-77995e302897.png)

Reply:
![image](https://user-images.githubusercontent.com/67829287/210584876-db462856-6e7e-4d84-934c-1acd40c41876.png)

Reply:
Copying over from the slack for record keeping

```
The difference between data_url and b0 in the simple unit tests is that
b0 is specified as an artifact we want to generate as well so we can get its value from the task that generates it def get_b0():
but data_url is a parameter , note how it is specified as
    input_parameters=['data_url'],
in the to_pipeline call, not an artifact
import os
directory = lineapy.to_pipeline(
    ["cleaned_data_housing", "linear_model_housing"],
    input_parameters=['data_url'],
    framework="DVC",
    pipeline_name="data_housing_pipeline",
    dependencies={ "linear_model_housing": { "cleaned_data_housing" } },
    pipeline_dag_config={"dag_flavor":"StagePerArtifact"}, 
    output_dir="./dvc/dags/"
)
which we assume to mean the user may want to parametrize this and provide a different value for data_url in their pipeline. We dont have a task to generate the value for this and rely on the orchestrator and user to provide us the value.


data_url is a user specified parameter
I think we should use BasePipelineWriter's get_pipeline_args

which returns the information needed to generate a params.yaml file, including a default value to use for the parameter
```

Reply:
Thanks for your explanations, I will work on this. I think we are pretty close to a perfect solution.

Reply:
👍 💯

Reply:
Fixed to generate `params.yaml` now too.